src/jit/simdcodegenxarch.cpp

```diff
@@ -475,9 +476,9 @@ instruction CodeGen::getOpForSIMDIntrinsic(SIMDIntrinsicID intrinsicId, var_type
 // Notes:
 //    This is currently only supported for floating point types.
 //
-void CodeGen::genSIMDScalarMove(var_types type, regNumber targetReg, regNumber srcReg, SIMDScalarMoveType moveType)
+void CodeGen::genSIMDScalarMove(
+    var_types targetType, var_types type, regNumber targetReg, regNumber srcReg, SIMDScalarMoveType moveType)
```

Comment:
I find it confusing to have both targetType and type parameters.  It would be better to name params targetType/type as simdType/baseType respectively.

Besides genSIMDScalarMove() is selecting instruction based on 'type' parameter but is passing targetType, which doesn't seem correct to me.

I still think in all the places where instruction selection is based on 'type' should use type/emitTypeSize(type) instead of targetType/emitTypeSize(targetType).